packages/flutter/lib/src/widgets/editable.dart

```diff
@@ -430,3 +463,153 @@ class _EditableLineWidget extends LeafRenderObjectWidget {
     return new TextSpan(style: style, text: text);
   }
 }
+
+enum _TextSelectionHandleSide { left, right }
+
+class _TextSelectionHandleOverlay extends StatefulWidget {
+  _TextSelectionHandleOverlay({
+    Key key,
+    this.onSelectionHandleChanged,
+    this.selection,
+    this.renderObject
+  }) : super(key: key);
+
+  final TextSelection selection;
+  final ValueChanged<TextSelection> onSelectionHandleChanged;
+  // TODO(mpcomplete): what if the renderObject is removed or replaced, or
+  // moves? Not sure what cases I need to handle, or how to handle them.
+  final RenderEditableLine renderObject;
+
+  @override
+  _TextSelectionHandleOverlayState createState() => new _TextSelectionHandleOverlayState();
+}
+
+/// The state for an overlay that displays 2 selection handles. This is intended
+/// to be used only when there is text selected (e.g. the TextSelection is not
+/// collapsed).
+class _TextSelectionHandleOverlayState extends State<_TextSelectionHandleOverlay> {
+  TextSelection selection;
+  List<Point> endpoints;
+
+  @override
+  void initState() {
+    super.initState();
+    selection = config.selection;
+    endpoints = config.renderObject.getEndpointsForSelection(selection);
+  }
+
+  @override
+  void didUpdateConfig(_TextSelectionHandleOverlay oldConfig) {
+    setState(() {
+      selection = config.selection;
+      endpoints = config.renderObject.getEndpointsForSelection(selection);
+    });
+  }
+
+  Point _dragPosition;
+  void _dragStart(Point position, _TextSelectionHandleSide side) {
+    _dragPosition = position;
+  }
+
+  void _dragUpdate(double delta, _TextSelectionHandleSide side) {
+    _dragPosition = _dragPosition + new Offset(delta, 0.0);
+    TextPosition position = config.renderObject.getPositionForPoint(_dragPosition);
+
+    TextSelection newSelection;
+    if (side == _TextSelectionHandleSide.left) {
+      newSelection = new TextSelection(
+        baseOffset: position.offset,
+        extentOffset: selection.extentOffset
+      );
+    } else {
+      newSelection = new TextSelection(
+        baseOffset: selection.baseOffset,
+        extentOffset: position.offset
+      );
+    }
+
+    if (newSelection.baseOffset >= newSelection.extentOffset)
+      return;
+
+    setState(() {
+      selection = newSelection;
+      endpoints = config.renderObject.getEndpointsForSelection(selection);
+      config.onSelectionHandleChanged(selection);
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Stack(
```

Comment:
Rather than creating a stack here, we should probably have two entries in the overlay, one for each of the selection handles.  When they move, they can call back into to the RawInputLineState which can merge the data, update the selection, and mark the entries as needing to rebuild.  Each selection handle can then operate independently.

The way you've got this set up currently doesn't handle multitouch properly (e.g., if one finger is dragging one handle at the same time as another finger is dragging another handle).  They'll fight over the _dragPosition.  If you set this up as two separate widgets with two separate overlay entries, you'll get multitouch correctness for free.